Quantum-Control-Applications/Superconducting/Single Fixed Transmon/Use Case - icognito - Optimized readout with optimal weights/README.md

```diff
@@ -0,0 +1,99 @@
+
+The readout pulse is `380 ns` long.
+
+## Methods and results
+
+The set of scripts to obtain the readout optimal weights are `TwoStateDiscriminator.py`,
+`TimeDiffCalibrator.py`, and `StateDiscriminator.py`. These three files work together during
+the experiment. Inside these files there are Python classes that are used during
+the training, benchmarking, and later usage in other QUA programs.
+
+First, we will briefly describe each of these files. `TimeDiffCalibrator.py` calculates
+the `time_diff` variable that we used to demodulate analog signals (in post-processing) 
+to match the values of demodulation in the FPGA. `TwoStateDiscriminator.py` contains
+functions to update the `configuration` and to perform measurements with the optimal weights.
+Last, `StateDiscriminator.py` contains the `train` function that from measured analog signals
+calculates the optimal weights.
+
+### Step 1: Training
+
+This steps performs measurements of analog signals when the qubit is in the ground and excited
+state. Once the analog signals are saved in the client PC, demodulation from `IF` to baseband
+is performed with functions defined inside of `StateDiscriminator.py`, and then the optimal weights
+are derived from `|g> - |e>` time traces. The program that we
+specifically used for training was `IQ_blobs_opt_weights_train.py` and `configuration.py`.
+Here we will show some results from the training.
+
+The image below was generated using the analysis tool [`two_state_discriminator`](https://github.com/qua-platform/py-qua-tools/tree/main/qualang_tools/analysis).
+In the *original data* subplot we can see the IQ blobs from the `|g> (blue)` and `|e> (orange)` demodulated
+in the FPGA. The `two_state_discriminator` further analyses the measured data, it rotates
+the blobs to project the separation between `|g>` and `|e>` in the `I-axis`. The analysis tool
+also displays the *1D Histogram* (after rotation). The Fidelity's are reported in the *confusion matrix*
+The readout fidelity is calculated as the average of `gg` and `ee` fidelities, thus
+resulting in a readout fidelity of `91.3 %`.
+
+![constant_weights](IQblobs_constant_weights.png)
+
+As mentioned above, we obtain the *time traces* of the ground and excited state in the
+trainin phase. We can visualize the `I` and `Q` time traces in the image below. Note 
+that the readout only lasts `380 ns`. The rightmost, subplot displays the optimal weights
+defined as `|g> - |e>`, and we can see that in the original data most of the information
+is already encoded in `I`. These derived optimal weights are the ones reintroduced in the
+`configuration` so that when demodulation is performed the serve the mask for the integration stage.
+From the third subplot we can become aware that in the first `100 ns` there is almost no
+separation between the `|g>` and `|e>` states.
+
+![time_traces](time_traces_and_opt_weights.png)
+
+There are two advantages of using *optimal weights*. First, when there is little to zero
```

Comment:
Very nice.
Actually, this figure also shows us there is probably more information if we continue demodulation. We expect the information always to start **and end** at 0, but here it ends at a finite (and increasing) value.
I would probably mention this, even if we cannot redo the measurement with longer demodulation (increase smearing and maybe ToF for the training part).